raiden/tests/integration/network/transport/rif_comms/test_client.py

```diff
@@ -1,36 +1,47 @@
+from typing import List, Dict
+
 import pytest
+from eth_utils import to_canonical_address
 from grpc._channel import _InactiveRpcError
 
 from raiden.tests.integration.network.transport.rif_comms.node import Node as CommsNode, Config as CommsConfig
 from raiden.tests.integration.network.transport.utils import generate_address
+from transport.rif_comms.client import Client
+from transport.rif_comms.cluster import Cluster
 from transport.rif_comms.utils import notification_to_payload, get_sender_from_notification
 
 
 @pytest.fixture()
-@pytest.mark.parametrize("amount_of_nodes")
-def comms_nodes(amount_of_nodes) -> {int, CommsNode}:
-    nodes = {}
+@pytest.mark.parametrize("nodes_to_clients")
+def comms_clients(nodes_to_clients: dict) -> Dict[int, Client]:
+    """
+    The cluster dict has the following structure:
+
+    {
+        1: 3 // a node with 3 clients connected
+        2: 1 // a node with a client
+        3: 2 // another node with 2 clients connected
+    }
 
-    # setup
-    for i in range(1, amount_of_nodes + 1):
-        node = CommsNode(CommsConfig(i))
-        nodes[i] = node
+    Keys must be in increasing order from 1 to N
+    Values are the amount of clients connected to that node.
+    """
 
-    yield nodes
+    cluster = Cluster(nodes_to_clients)
+    yield cluster.get_clients()
 
     # teardown
-    for node in nodes.values():
-        node.stop()
+    cluster.shutdown()
 
 
 @pytest.mark.xfail(reason="wrong exception message from comms node")
 def test_connect():
     # the comms_nodes fixture is not used to prevent automatic connect
     nodes = []
     try:
-        node = CommsNode(CommsConfig(node_number=1, auto_connect=False))
+        node = CommsNode(CommsConfig(node_number=1, amount_of_clients=1, auto_connect=False))
         nodes.append(node)
-        client, address = node.client, node.address
+        client, address = node.clients[0], node.clients[0].rsk_address.address
```

Comment:
since we went through all the trouble of starting indexes at `1` for every other collection in these tests, it would be nice for `clients` to also be like this.

it's a NTH, not a blocker though

Reply:
we dont need to access trought index now since we iterate over the clients and the cluster just gives us a flattened list

this test case doesnt use the fixture, so i dont find useful to change the way is access to the first element of a list

Reply:
sure, makes sense